pkg/diagnostics/analyze.go

```diff
@@ -0,0 +1,319 @@
+package diagnostics
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	nbv1 "github.com/noobaa/noobaa-operator/v5/pkg/apis/noobaa/v1alpha1"
+	"github.com/noobaa/noobaa-operator/v5/pkg/backingstore"
+	"github.com/noobaa/noobaa-operator/v5/pkg/bundle"
+	"github.com/noobaa/noobaa-operator/v5/pkg/namespacestore"
+	"github.com/noobaa/noobaa-operator/v5/pkg/options"
+	"github.com/noobaa/noobaa-operator/v5/pkg/util"
+
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/spf13/cobra"
+)
+
+// RunAnalyzeBackingStore runs a CLI command
+func RunAnalyzeBackingStore(cmd *cobra.Command, args []string) {
+	backingStore := backingstore.GetBackingStoreFromArgs(cmd, args)
+	util.Logger().Printf("⏳ Running this command will run tests on BackingStore %s...\n", backingStore.Name)
+	collector := collectorInstance(cmd)
+	makeDirForLogs(collector)
+	analyzeBackingstore(cmd, backingStore, collector)
+	destDir, _ := cmd.Flags().GetString("dir")
+	collector.ExportDiagnostics(destDir)
+	collector.log.Println("✅ Logs were saved")
```

Comment:
I suggest printing the path of the saved logs. same for the db-dump and diagnostics